### PR TITLE
Plugin E2E: Add contributing guidelines

### PR DESCRIPTION
**Pull request by sunker**
**What this PR does / why we need it**:

There are two purposes of this PR:
* add contributing guidelines for this workspace
* in the contributing guidelines, add a section that describe in detail how to fix a broken scenario, test it with grafana/grafana and deploy it to NPM. 

For the latter of the two, the scenario will be: 
A contributor makes a change in grafana/grafana, and that changes causes Playwright tests that are using fixtures, models and expect matchers defined in this package to break. The contributor will then have provide a fix in this repo, even though their knowledge of this package may be zero or close to zero. Therefore, these instructions needs to be as clear as possible. 

I'm working on a corresponding PR in grafana/grafana that will link to this section of the contributing guidelines in the case a Playwright test fails in core Grafana. 

**Which issue(s) this PR fixes**:

Part of https://github.com/grafana/plugin-tools/issues/576

Fixes #

**Special notes for your reviewer**:

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @grafana/plugin-e2e@0.7.0-canary.634.ef4697c.0
  # or 
  yarn add @grafana/plugin-e2e@0.7.0-canary.634.ef4697c.0
  ```
</details>

### CODE REVIEW

**Review by academo — APPROVED** [review]
LGTM but wait for @josmperez to take a look

**Review by sympatheticmoose — COMMENTED** [review]
❤️

File: packages/plugin-e2e/CONTRIBUTING.md [hunk 1]
```diff
@@ -1 +1,142 @@
...
+
+### Installing
+
+```bash
+git clone git@github.com:grafana/plugin-tools.git
+cd plugin-tools
+npm install
+
+#Each version of Playwright needs specific versions of browser binaries to operate. You will need to use the Playwright CLI to install these browsers.
+yarn playwright install
```
[the remark this answers is not in the document]
Comment (sunker) [reply packages/plugin-e2e/CONTRIBUTING.md:18]: Oh, no we should only be using npm in this repo. Good catch!

File: packages/plugin-e2e/CONTRIBUTING.md [hunk 2]
```diff
@@ -1 +1,142 @@
...
+2. Run Playwright tests locally - `npm run playwright:test`
+
+3. Push the changes in your local PR, create a draft PR in [grafana/plugin-tools](https://github.com/grafana/plugin-tools/) and add the labels `release` and `minor|patch`. CI will run all Playwright tests against a set of different Grafana versions. If not all of them pass, it may be because you've introduced a change that is no longer compatible with older versions of Grafana.
+
+4. Once CI passes, `auto` will publish a canary release to NPM. You can find the version number at the bottom of the PR description.
+
+5. Install the pre-release of @grafana/plugin-e2e in [grafana/grafana](https://github.com/grafana/grafana) and run Playwright tests.
+
+```bash
+yarn add @grafana/plugin-e2e@0.3.0-canary.623.aedff75.0
```
[the remark this answers is not in the document]
Comment (sunker) [reply packages/plugin-e2e/CONTRIBUTING.md:136]: This is a bit messy, but grafana core uses yarn so this is correct.

File: packages/plugin-e2e/CONTRIBUTING.md [hunk 3]
```diff
@@ -1 +1,142 @@
...
+yarn e2e:plugin
+```
+
+6. If all tests pass in `grafana/grafana` and the PR in `grafana/plugin-tools` is approved, you can go ahead and merge the PR.
```
[the remark this answers is not in the document]
Comment (sunker) [reply packages/plugin-e2e/CONTRIBUTING.md:140]: That is true. I'll fix

**Review by josmperez — APPROVED** [review]
LGTM.

**Pull request merged**